src/interfaces/earnings.ts

```diff
@@ -31,9 +31,14 @@ const BigZero = new BigNumber(0);
 
 export class EarningsInterface<C extends ChainId> extends ServiceInterface<C> {
   async protocolEarnings(): Promise<String> {
-    const response = (await this.yearn.services.subgraph.fetchQuery(PROTOCOL_EARNINGS)) as ProtocolEarningsResponse;
+    const response = await this.yearn.services.subgraph.fetchQuery<ProtocolEarningsResponse>(PROTOCOL_EARNINGS);
 
     let result = BigZero;
+
+    if (!response?.data || !response.data?.vaults) {
+      return result.toFixed(0);
+    }
```

Comment:
Same reasoning as https://github.com/yearn/yearn-sdk/pull/223/files#r807530217 -- if no `data` just return 0

Reply:
shouldn't we actually throw an error?

Reply:
As I wasn't 100% sure I didn't want to change the behaviour, we could throw but that would change the way this function was working before. Happy to throw, but would need someone with more context to make sure that's what we want

Reply:
I think when I was writing it I wanted the earnings code to fail gracefully, however it probably makes more sense to throw an error here, and for it to be handled by the consumer. I don't think we should change it though as it might break consumers' existing functionality. 

Also, I think this method should eventually be removed. We're not able to accurately calculate the total protocol earnings from the subgraph due to technical limitations, it's better handled by https://github.com/yearn/yearn-exporter. Maybe we could add a deprecation notice to this function in JSDoc, saying that the values are not accurate and it'll be removed in the future